### ISSUE TITLE
Support for "allowTypeImports" with "patterns" in `no-restricted-imports`

### ISSUE THREAD

**Issue by devklepacki** [eslint/eslint#16791]
### What rule do you want to change?

no-restricted-imports

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
'no-restricted-imports': [
  'error', {
    patterns: [{
      group: ['server/*'],
      message: 'Only type definition imports are allowed from server directory',
      allowTypeImports: true,
    }],
  },
],
```

### What does the rule currently do for this code?

Restricts all imports from pattern's group

### What will the rule do after it's changed?

The rule will allow for type definition imports similarly to `paths`  option

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

On `paths` option there's support for `allowTypeImports` (https://typescript-eslint.io/rules/no-restricted-imports/#allowtypeimports). With monorepo tools and others like tRPC this could be useful to add the support for `allowTypeImports` for `pattern` option. This'd make extra sure that no server-side code is imported into the client bundle.

**Comment by mdjermanovic** [comment]
Hi @devklepacki, thanks for the issue!

Core rules don't support typescript-specific options. 

> On `paths` option there's support for `allowTypeImports` (https://typescript-eslint.io/rules/no-restricted-imports/#allowtypeimports). With monorepo tools and others like tRPC this could be useful to add the support for `allowTypeImports` for `pattern` option. This'd make extra sure that no server-side code is imported into the client bundle.

This is a typescript-eslint plugin rule `@typescript-eslint/no-restricted-imports`. If you'd like to propose a new option for this rule, you can open an issue in the typescript-eslint repository:

https://github.com/typescript-eslint/typescript-eslint

**Comment by devklepacki** [comment]
Ah, thank you, and sorry!
